Indicators/Vortex.cs

```diff
@@ -0,0 +1,132 @@
+/// The values are accumulated and smoothed using a rolling window of the specified period.
+/// </summary>
+
+public class VortexIndicator : IndicatorBase<IBaseDataBar>
+{
+    private readonly int _period;
+    private readonly RollingWindow<IBaseDataBar> _window;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="VortexIndicator"/> class.
+    /// </summary>
+    /// <param name="period">The number of periods used to calculate the indicator.</param>
+    public VortexIndicator(int period) : base("VTX")
+    {
+        _period = period;
+        _window = new RollingWindow<IBaseDataBar>(period + 1);
+    }
+
+    /// <summary>
+    /// Computes the next value of the Vortex Indicator.
+    /// </summary>
+    /// <param name="input">The input data point.</param>
+    /// <returns>The value of the Vortex Indicator.</returns>
+    protected override decimal ComputeNextValue(IBaseDataBar input)
+    {
+        _window.Add(input);
+
+        if (_window.Count < _period + 1)
+        {
+            return 0m; // or possibly Decimal.MinValue to indicate insufficient data
+        }
+
+        decimal sumTR = 0m;
+        decimal sumVMPlus = 0m;
+        decimal sumVMMinus = 0m;
+
+        // Starting from 1 since we want to compare current bar with the previous bar
+        for (int i = 1; i < _window.Count; i++)
+        {
+            var current = _window[i - 1]; // current bar
+            var previous = _window[i]; // previous bar
+
+            decimal tr = Math.Max(current.High - current.Low, Math.Max(Math.Abs(current.High - previous.Close), Math.Abs(current.Low - previous.Close)));
+            decimal vmPlus = Math.Abs(current.High - previous.Low);
+            decimal vmMinus = Math.Abs(current.Low - previous.High);
+
+            sumTR += tr;
+            sumVMPlus += vmPlus;
+            sumVMMinus += vmMinus;
+        }
+
+        decimal positiveVortex = sumVMPlus / sumTR;
+        decimal negativeVortex = sumVMMinus / sumTR;
+
+        PositiveVortexIndicator.Update(input.Time, positiveVortex);
+        NegativeVortexIndicator.Update(input.Time, negativeVortex);
+
+        // Depending on your test setup, you might want to return different values:
+        return positiveVortex; // If the test expects Positive Vortex
+        // return negativeVortex; // If the test expects Negative Vortex
+    }
+
+    /// <summary>
+    /// Gets the positive vortex indicator.
+    /// </summary>
+    public IndicatorBase<IndicatorDataPoint> PositiveVortexIndicator { get; } = new Identity("PositiveVortex");
+
+    /// <summary>
+    /// Gets the negative vortex indicator.
+    /// </summary>
+    public IndicatorBase<IndicatorDataPoint> NegativeVortexIndicator { get; } = new Identity("NegativeVortex");
+
+    /// <summary>
+    /// Gets a value indicating whether this indicator is ready and fully initialized.
+    /// </summary>
+    public override bool IsReady => _window.Count == _period + 1;
```

Comment:
Another common practice in Lean is having properties and fields at the top of the class, privates first